src/pages/workspace/companyCards/utils.tsx

```diff
@@ -48,49 +47,82 @@ function getExportMenuItem(
+                    exportType = CONST.COMPANY_CARDS.EXPORT_CARD_TYPES.NVP_QUICKBOOKS_ONLINE_EXPORT_ACCOUNT;
                     break;
-                case CONST.QUICKBOOKS_NON_REIMBURSABLE_EXPORT_ACCOUNT_TYPE.DEBIT_CARD:
+                }
+                case CONST.QUICKBOOKS_NON_REIMBURSABLE_EXPORT_ACCOUNT_TYPE.DEBIT_CARD: {
                     data = quickbooksOnlineBankAccounts ?? [];
-                    title = companyCard?.nameValuePairs?.quickbooks_online_export_account_debit ?? nonReimbursableExpensesAccount?.name;
+                    isDefaultTitle = !!(
+                        defaultAccount &&
+                        (!companyCard?.nameValuePairs?.quickbooks_online_export_account_debit ||
+                            companyCard?.nameValuePairs?.quickbooks_online_export_account_debit === CONST.COMPANY_CARDS.DEFAULT_EXPORT_TYPE)
+                    );
+                    title = isDefaultTitle ? defaultCard : companyCard?.nameValuePairs?.quickbooks_online_export_account_debit;
+                    selectedAccount = companyCard?.nameValuePairs?.quickbooks_online_export_account_debit ?? defaultAccount;
                     exportType = CONST.COMPANY_CARDS.EXPORT_CARD_TYPES.NVP_QUICKBOOKS_ONLINE_EXPORT_ACCOUNT_DEBIT;
                     break;
+                }
                 default:
                     shouldShowMenuItem = false;
                     data = [];
             }
 
+            const resultData = data.length > 0 ? [...data, defaultMenuItem] : data;
+
             return {
                 description,
                 title,
                 exportType,
                 shouldShowMenuItem,
                 onExportPagePress: () => Navigation.navigate(ROUTES.POLICY_ACCOUNTING_QUICKBOOKS_ONLINE_EXPORT.getRoute(policyID)),
-                data: data.map((card) => ({
+                data: resultData.map((card) => ({
                     value: card.name,
                     text: card.name,
                     keyForList: card.name,
-                    isSelected: card.name === title,
+                    isSelected: isDefaultTitle ? card.name === defaultCard : card.name === selectedAccount,
                 })),
             };
         }
         case CONST.POLICY.CONNECTIONS.NAME.XERO: {
             const type = translate('workspace.xero.xeroBankAccount');
             const description = currentConnectionName && type ? translate('workspace.moreFeatures.companyCards.integrationExport', {integration: currentConnectionName, type}) : undefined;
             const exportType = CONST.COMPANY_CARDS.EXPORT_CARD_TYPES.NVP_XERO_EXPORT_BANK_ACCOUNT;
-            const selectedAccount = (bankAccounts ?? []).find((bank) => bank.id === exportConfiguration?.nonReimbursableAccount);
+            const defaultAccount = exportConfiguration?.nonReimbursableAccount;
+            const isDefaultTitle = !!(
+                defaultAccount &&
```

Comment:
When selecting the `default card`, the BE doesn't return exportConfiguration?.nonReimbursableAccount

Reply:
<img width="536" alt="Screenshot 2024-11-22 at 15 12 34" src="https://github.com/user-attachments/assets/ee6955a8-df84-4177-8509-a884f39e5120">

Reply:
In this case, it seems we haven't set up the Xero Bank Account, so default card refers to undefined

Reply:
On Xero Connection, we still display the Xero Bank Account value 

<img width="842" alt="Screenshot 2024-11-22 at 15 25 14" src="https://github.com/user-attachments/assets/29e5599b-62e9-4be0-aff4-8736bf21e3a3">

because we already set fallback value bankAccounts[0]

https://github.com/Expensify/App/blob/5f483dc37ccfbb967d8eeb49302d3dfa010f93e9/src/pages/workspace/accounting/xero/export/XeroExportConfigurationPage.tsx#L26


Thus, We can do the same thing on card detail page